### PR TITLE
Improve 404 error handling

### PR DESCRIPTION
**Pull request by tommy-mitchell**
Similar to https://github.com/sindresorhus/npm-user/pull/23

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -1,11 +1,31 @@
...
 	t.is(await npmEmail('sindresorhus'), 'sindresorhus@gmail.com');
 	t.is(await npmEmail('vdemedes'), 'vadimdemedes@hey.com');
 });
+
+test.serial('non-existent user', async t => {
+	t.is(await npmEmail('nnnope'), undefined);
+});
+
+test.serial('handles 404', async t => {
+	globalThis.fetch = async () => {
```
Comment (sindresorhus) [inline test.js:21]: You need to reset this back at the end of the test.

File: test.js [hunk 2]
```diff
@@ -1,11 +1,31 @@
 import test from 'ava';
+import {HTTPError} from 'ky';
 import npmEmail from './index.js';
 
-test('invalid input', async t => {
+test.serial('invalid input', async t => {
```
Comment (sindresorhus) [inline test.js]: No reason these have to be serial.

File: index.js [hunk 3]
```diff
@@ -8,7 +8,11 @@ export default async function npmEmail(username) {
 	const url = new URL(`https://registry.npmjs.com/-/v1/search?&text=maintainer:${encodeURIComponent(username)}`);
 
 	try {
-		const {objects: results} = await ky(url).json();
+		const {objects: results, total} = await ky(url).json();
+
+		if (total === 0) {
+			return undefined;
```
Comment (sindresorhus) [inline index.js]:
```suggestion
			return;
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: test.js (hunk 2 above)
Comment (tommy-mitchell) [reply test.js]: Yeah, I was being lazy to avoid resetting the `fetch` mock lol. Will update.

**Comment by tommy-mitchell** [comment]
Updated.

File: test.js [hunk 4]
```diff
@@ -9,3 +10,24 @@ test('valid username', async t => {
 	t.is(await npmEmail('sindresorhus'), 'sindresorhus@gmail.com');
 	t.is(await npmEmail('vdemedes'), 'vadimdemedes@hey.com');
 });
+
+test('non-existent user', async t => {
+	t.is(await npmEmail('nnnope'), undefined);
+});
+
+test('handles 404', async t => {
```
Comment (sindresorhus) [inline test.js]: I meant the other tests. This one does indeed need to be serial because of the mocking.

**Pull request merged**